app/components/DownloadMap/index.js

```diff
@@ -88,56 +91,60 @@ const DownloadMap = ({ navigation }) => {
-        const progressListener = (offlineRegion, status) => {
-          if (status.percentage == 100) {
-            createOfflineMap({
-              name: offllineMapId,
-              size: status.completedTileSize,
-              areaName: areaName,
-            })
-              .then(() => {
-                setIsLoaderShow(false);
-                setTimeout(() => alert(i18next.t('label.download_map_complete')), 1000);
-                getAllOfflineMapslocal();
-                setAreaName('');
+    if (netInfo.isConnected && netInfo.isInternetReachable) {
+      setIsLoaderShow(true);
+      let coords = await MapBoxGLRef.current.getCenter();
+      let bounds = await MapBoxGLRef.current.getVisibleBounds();
+      getAreaName({ coords })
+        .then(async (areaName) => {
+          setAreaName(areaName);
+          const progressListener = (offlineRegion, status) => {
+            if (status.percentage == 100) {
+              createOfflineMap({
+                name: offllineMapId,
+                size: status.completedTileSize,
+                areaName: areaName,
               })
-              .catch((err) => {
-                setIsLoaderShow(false);
-                setAreaName('');
-                alert(i18next.t('label.download_map_area_exists'));
-              });
-          }
-        };
-        const errorListener = (offlineRegion, err) => {
-          if (err.message !== 'timeout') {
-            setIsLoaderShow(false);
-            setAreaName('');
-            alert(err.message);
-          }
-        };
-        await MapboxGL.offlineManager.createPack(
-          {
-            name: offllineMapId,
-            styleURL: 'mapbox://styles/sagararl/ckdfyrsw80y3a1il9eqpecoc7',
-            minZoom: 14,
-            maxZoom: 20,
-            bounds: bounds,
-          },
-          progressListener,
-          errorListener,
-        );
-      })
-      .catch((err) => {
-        setIsLoaderShow(false);
-        setAreaName('');
-        alert(i18next.t('label.download_map_area_failed'));
-      });
+                .then(() => {
+                  setIsLoaderShow(false);
+                  setTimeout(() => alert(i18next.t('label.download_map_complete')), 1000);
+                  getAllOfflineMapslocal();
+                  setAreaName('');
+                })
+                .catch((err) => {
+                  setIsLoaderShow(false);
+                  setAreaName('');
+                  alert(i18next.t('label.download_map_area_exists'));
+                });
+            }
+          };
+          const errorListener = (offlineRegion, err) => {
+            if (err.message !== 'timeout') {
+              setIsLoaderShow(false);
+              setAreaName('');
+              alert(err.message);
+            }
+          };
+          await MapboxGL.offlineManager.createPack(
+            {
+              name: offllineMapId,
+              styleURL: 'mapbox://styles/sagararl/ckdfyrsw80y3a1il9eqpecoc7',
```

Comment:
@sagararyal are you aware this is a production asset that won't get changed accidentally any more?